testing/adios2/engine/bp/CMakeLists.txt

```diff
@@ -4,6 +4,10 @@
 # ------------------------------------------------------------------------------#
 include(ADIOSFunctions)
 
+if(ADIOS2_HAVE_CUDA OR Kokkos_ENABLE_CUDA)
```

Comment:
I think that in detectoptions `Kokkos_ENABLE_CUDA` should imply `ADIOS2_HAVE_CUDA`

Reply:
They are not the same thing, `ADIOS2_HAVE_CUDA` is saying ADIOS2 has the CUDA backend enabled. 

`Kokkos_ENABLE_CUDA` is saying ADIOS2 has the Kokkos backend enabled and KOKKOS has CUDA enabled.

I can change the name to `ADIOS2_HAVE_{name}_BACKEND` with name=`CUDA` or 'KOKKOS`

We already have a `ADIOS2_HAVE_GPU_SUPPORT` to know if any of the backends can handle GPU pointers. On a CUDA machine this is equivalent to knowing if CUDA is enabled in ADIOS2 (either through the CUDA backend or through the Kokkos backend).